controller/workitem_blackbox_test.go

```diff
@@ -148,6 +154,188 @@ func (s *WorkItemSuite) TestCreateWI() {
+
+	_, result2 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.controller, &payload)
+	_, result3 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.controller, &payload)
+	payload2 := minimumRequiredReorderPayload()
+
+	var dataArray []*app.WorkItem2 // dataArray contains the workitem(s) that have to be reordered
+	dataArray = append(dataArray, result3.Data)
+	payload2.Data = dataArray
+	payload2.Position.ID = *result2.Data.ID // Position.ID specifies the workitem ID above or below which the workitem(s) should be placed
+	payload2.Position.Direction = above
+	_, reordered1 := test.ReorderWorkitemOK(s.T(), s.svc.Context, s.svc, s.controller, &payload2) // Returns the workitems which are reordered
+
+	require.Len(s.T(), reordered1.Data, 1) // checks the correct number of workitems reordered
+	assert.Equal(s.T(), result3.Data.Attributes["version"].(int)+1, reordered1.Data[0].Attributes["version"])
+	assert.Equal(s.T(), *result3.Data.ID, *reordered1.Data[0].ID)
+}
+
+// TestReorderBelow is positive test which tests successful reorder by providing valid input
+// This case reorders one workitem -> result1 and places it **below** result1
+func (s *WorkItemSuite) TestReorderWorkitemBelowOK() {
+	payload := minimumRequiredCreateWithType(workitem.SystemBug)
+	payload.Data.Attributes[workitem.SystemTitle] = "Reorder Test WI"
+	payload.Data.Attributes[workitem.SystemState] = workitem.SystemStateClosed
+
+	_, result1 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.controller, &payload)
+	_, result2 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.controller, &payload)
+
+	// This workitem is created but not used to clearly demonstrate that the reorder workitem is moved between **two** workitems i.e. result2 and result3 and not to the **bottom** of the list
+	test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.controller, &payload)
+
+	payload2 := minimumRequiredReorderPayload()
+
+	var dataArray []*app.WorkItem2 // dataArray contains the workitem(s) that have to be reordered
+	dataArray = append(dataArray, result1.Data)
+	payload2.Data = dataArray
+	payload2.Position.ID = *result2.Data.ID // Position.ID specifies the workitem ID above or below which the workitem(s) should be placed
+	payload2.Position.Direction = below
+
+	_, reordered1 := test.ReorderWorkitemOK(s.T(), s.svc.Context, s.svc, s.controller, &payload2) // Returns the workitems which are reordered
+
+	require.Len(s.T(), reordered1.Data, 1) // checks the correct number of workitems reordered
+	assert.Equal(s.T(), result1.Data.Attributes["version"].(int)+1, reordered1.Data[0].Attributes["version"])
+	assert.Equal(s.T(), *result1.Data.ID, *reordered1.Data[0].ID)
+}
+
+// TestReorderTop is positive test which tests successful reorder by providing valid input
+// This case reorders one workitem -> result2 and places it to the top of the list
+func (s *WorkItemSuite) TestReorderWorkitemTopOK() {
```

Comment:
Right now, the request payload looks like this:

Assume you want to move workitem 5 to top i.e. above workitem 1, then:

curl -i -X PATCH localhost:8080/api/workitems/reorder -H "Authorization: Bearer " -d '{"data": [{"type": "workitems", "attributes":{"version":0}, "id": "5"}], "position": {"id":"1", "direction":"above"}}'

When you said -> 

> move C to Top (not before A)

Should the request payload also change to depict that an item is moved to top and not above some item (in this case 1)?